flink-streaming-java/src/main/java/org/apache/flink/streaming/api/operators/sort/MultiInputSortingDataInput.java

```diff
@@ -315,23 +342,40 @@ private InputStatus addNextToQueue(HeadElement reuse, DataOutput<IN> output) thr
 	 * element of all sorting inputs. Should be used by the {@link StreamInputProcessor} to
 	 * choose the next input to consume from.
 	 */
-	private static class InputSelector implements InputSelectable {
+	private static class InputSelector implements InputSelectable, BoundedMultiInput {
 
 		private final CommonContext commonContext;
-		private final int numberOfInputs;
+		private final int numInputs;
+		private final ConcurrentLinkedQueue<Integer> passThroughInputsIndices;
```

Comment:
Do we need the `Concurrent` queue here? If I am not mistaken all of it happens from within the main mailbox thread.

Reply:
Could be true, I went for a safe thing but could remove it. 👌